src/utils/wallet.js

```diff
@@ -101,6 +103,55 @@ export class Wallet {
       return await this.getAccount(this.accountId).getAccessKeys()
    }
 
+   async connectWapm(wamp) {
+      try {
+         return await new Promise((resolve, reject) => {
+            wamp.onopen = session => resolve(session)
+            wamp.onclose = reason => reject(reason)
+            wamp.open()
+         });
+      } catch (error) {
+         console.error('Connection failure due to:', error)
```

Comment:
why is error swallowed instead of being returned to caller?

Reply:
yes, I will change it